File: cdktf-examples/java/src/main/java/com/mycompany/constructs/SimpleKubernetesWebApp.java
package com.mycompany.constructs;

import org.jetbrains.annotations.NotNull;

import com.hashicorp.cdktf.TerraformOutput;
import com.hashicorp.cdktf.TerraformOutputConfig;

import software.constructs.Construct;

public class SimpleKubernetesWebApp extends Construct {
    public final KubernetesWebAppDeployment deployment;
    public final KubernetesNodePortService service;
    public final SimpleKubernetesWebAppConfig config;

    public SimpleKubernetesWebApp(@NotNull Construct scope, @NotNull String id,
            @NotNull SimpleKubernetesWebAppConfig config) {
        super(scope, id);

        this.config = config;

        this.deployment = new KubernetesWebAppDeployment(this, "deployment", new KubernetesWebAppDeploymentConfig()
                .setImage(config.image())
                .setReplicas(config.replicas())
                .setApp(config.app())
                .setComponent(config.component())
                .setEnvironment(config.environment())
                .setEnv(config.env()));

        this.service = new KubernetesNodePortService(this, "service", new KubernetesNodePortServiceConfig()
                .setApp(config.app())
                .setComponent(config.component())
                .setEnvironment(config.environment())
                .setPort(config.port()));

        new TerraformOutput(this, "url", TerraformOutputConfig.builder()
                .value(String.format("http://localhost:%s", config.port())).build());
    }

}

File: app/constructs/SimpleKubernetesWebApp.cs
using System;
using System.IO;
using System.Collections.Generic;
using Constructs;
using HashiCorp.Cdktf;
using HashiCorp.Cdktf.Providers.Kubernetes.Deployment;


namespace MyCompany.MyApp
{

    class SimpleKubernetesWebAppConfig
    {
        public string Image { get; set; }
        public int Replicas { get; set; }
        public string App { get; set; }
        public string Component { get; set; }
        public string Environment { get; set; }
        public Dictionary<string, string> Env { get; set; }
        public int Port { get; set; }
    }

    class SimpleKubernetesWebApp : Construct
    {
        public SimpleKubernetesWebAppConfig Config { get; }
        public KubernetesWebAppDeployment Deployment { get; }
        public KubernetesNodePortService Service { get; }

        public SimpleKubernetesWebApp(Construct scope, string id, SimpleKubernetesWebAppConfig config) : base(scope, id)
        {

            this.Config = config;
            this.Deployment = new KubernetesWebAppDeployment(this, "Deployment", new KubernetesWebAppDeploymentConfig
            {
                Image = config.Image,
                Replicas = config.Replicas,
                App = config.App,
                Component = config.Component,
                Environment = config.Environment,
                Env = config.Env
            });

            this.Service = new KubernetesNodePortService(this, "Service", new KubernetesNodePortServiceConfig
            {
                Port = config.Port,
                App = config.App,
                Component = config.Component,
                Environment = config.Environment
            });

            new TerraformOutput(this, "url", new TerraformOutputConfig
            {
                Value = $"http://localhost:{config.Port}"
            });
        }
    }
}